src/storage/v2/disk/storage.cpp

```diff
@@ -0,0 +1,1703 @@
+    return std::nullopt;
+  }
+
+  const std::string value_str{value.ToString()};
+  const auto labels{utils::DeserializeLabelsFromLabelIndexStorage(value_str)};
+  return CreateVertex(index_accessor, gid, labels, utils::DeserializePropertiesFromLabelIndexStorage(value_str),
+                      index_delta);
+}
+
+std::optional<storage::VertexAccessor> DiskStorage::DiskAccessor::LoadVertexToLabelPropertyIndexCache(
+    const rocksdb::Slice &key, const rocksdb::Slice &value, Delta *index_delta,
+    utils::SkipList<storage::Vertex>::Accessor index_accessor) {
+  storage::Gid gid = Gid::FromUint(std::stoull(utils::ExtractGidFromLabelPropertyIndexStorage(key.ToString())));
+  if (VertexExistsInCache(index_accessor, gid)) {
+    return std::nullopt;
+  }
+
+  const std::string value_str{value.ToString()};
+  const auto labels{utils::DeserializeLabelsFromLabelPropertyIndexStorage(value_str)};
+  return CreateVertex(index_accessor, gid, labels,
+                      utils::DeserializePropertiesFromLabelPropertyIndexStorage(value.ToString()), index_delta);
+}
+
+std::optional<EdgeAccessor> DiskStorage::DiskAccessor::DeserializeEdge(const rocksdb::Slice &key,
+                                                                       const rocksdb::Slice &value) {
+  const auto edge_parts = utils::Split(key.ToStringView(), "|");
+  const Gid edge_gid = Gid::FromUint(std::stoull(edge_parts[4]));
+
+  auto edge_acc = edges_.access();
+  auto res = edge_acc.find(edge_gid);
+  if (res != edge_acc.end()) {
+    return std::nullopt;
+  }
+
+  const auto [from_gid, to_gid] = std::invoke(
+      [](const auto &edge_parts) {
+        if (edge_parts[2] == "0") {  // out edge
+          return std::make_pair(edge_parts[0], edge_parts[1]);
+        }
+        // in edge
+        return std::make_pair(edge_parts[1], edge_parts[0]);
+      },
+      edge_parts);
+
+  const auto from_acc = FindVertex(Gid::FromUint(std::stoull(from_gid)), View::OLD);
+  const auto to_acc = FindVertex(Gid::FromUint(std::stoull(to_gid)), View::OLD);
+  if (!from_acc || !to_acc) {
+    throw utils::BasicException("Non-existing vertices found during edge deserialization");
+  }
+  const auto edge_type_id = storage::EdgeTypeId::FromUint(std::stoull(edge_parts[3]));
+  auto maybe_edge = CreateEdge(&*from_acc, &*to_acc, edge_type_id, edge_gid, value.ToStringView(), key.ToString());
+  MG_ASSERT(maybe_edge.HasValue());
+
+  return *maybe_edge;
+}
+
+VerticesIterable DiskStorage::DiskAccessor::Vertices(View view) {
+  auto *disk_storage = static_cast<DiskStorage *>(storage_);
+  rocksdb::ReadOptions ro;
+  std::string strTs = utils::StringTimestamp(transaction_.start_timestamp);
+  rocksdb::Slice ts(strTs);
+  ro.timestamp = &ts;
+  auto it =
+      std::unique_ptr<rocksdb::Iterator>(disk_transaction_->GetIterator(ro, disk_storage->kvstore_->vertex_chandle));
+  for (it->SeekToFirst(); it->Valid(); it->Next()) {
+    LoadVertexToMainMemoryCache(it->key(), it->value());
+  }
```

Comment:
Can we return this as iterable, and then load vertices to main memory cache as something iterates over vertices?